src/backends/backend/onnxruntime/onnxruntime.cc

```diff
@@ -0,0 +1,1873 @@
+                    (std::string(
+                         "OpenVINO Execution Accelerator is set for '") +
+                     Name() + "' on CPU")
+                        .c_str());
+                continue;
+              }
+#endif  // TRITON_ENABLE_ONNXRUNTIME_OPENVINO
+              return TRITONSERVER_ErrorNew(
+                  TRITONSERVER_ERROR_INVALID_ARG,
+                  (std::string("unknown Execution Accelerator '") + name +
+                   "' is requested")
+                      .c_str());
+            }
+          }
+        }
+      }
+    }
+  }
+
+  // ONNX session creation with OpenVINO is not thread-safe,
+  // so multiple creations are serialized with a global lock.
+  static std::mutex global_context_mu;
+  std::unique_lock<std::mutex> glock(global_context_mu, std::defer_lock);
+  if (need_lock) {
+    glock.lock();
+  }
+
+  // Register all op libraries that contain custom operations.
+  {
+    ni::TritonJson::Value model_ops;
+    if (model_config_.Find("model_operations", &model_ops)) {
+      ni::TritonJson::Value op_library_filenames;
+      if (model_ops.Find("op_library_filename", &op_library_filenames)) {
+        for (size_t op_idx = 0; op_idx < op_library_filenames.ArraySize();
+             op_idx++) {
+          std::string op_filename;
+          RETURN_IF_ERROR(
+              op_library_filenames.IndexAsString(op_idx, &op_filename));
+          void* library_handle = nullptr;  // leak this, no harm.
+          RETURN_IF_ORT_ERROR(ort_api->RegisterCustomOpsLibrary(
+              soptions, op_filename.c_str(), &library_handle));
+        }
+      }
+    }
+  }
+
+  RETURN_IF_ERROR(OnnxLoader::LoadSession(
+      true /* is_path */, *model_path, soptions, session));
+  RETURN_IF_ORT_ERROR(ort_api->GetAllocatorWithDefaultOptions(allocator));
+
+  return nullptr;  // success
+}
+
+TRITONSERVER_Error*
+ModelState::AutoCompleteConfig()
+{
+  // If the model configuration already specifies max_batch_size,
+  // inputs or outputs then don't to any auto-completion.
+  size_t input_cnt = 0;
+  size_t output_cnt = 0;
+  {
+    ni::TritonJson::Value inputs;
+    if (ModelConfig().Find("input", &inputs)) {
+      input_cnt = inputs.ArraySize();
+    }
+    ni::TritonJson::Value outputs;
+    if (ModelConfig().Find("output", &outputs)) {
+      output_cnt = outputs.ArraySize();
+    }
+  }
+
+  if ((MaxBatchSize() > 0) || (input_cnt > 0) || (output_cnt > 0)) {
+    LOG_MESSAGE(
+        TRITONSERVER_LOG_INFO,
+        (std::string("skipping model configuration auto-complete for '") +
+         Name() + "': max_batch_size, inupts or outputs already specified")
+            .c_str());
+    return nullptr;  // success
+  }
+
+  std::string artifact_name;
+  RETURN_IF_ERROR(
+      ModelConfig().MemberAsString("default_model_filename", &artifact_name));
+
+  std::string model_path;
+  OrtSession* session;
+  OrtAllocator* allocator;
+  // FIXME no early return, must unload session
```

Comment:
Can use smart pointer to unload on return.

Reply:
Fixed